hstream-store/HStream/Store/Internal/Foreign.hs

```diff
@@ -105,6 +110,17 @@ retryWhileAgain f retries = do
     _ -> E.throwStreamError errno callStack
 {-# INLINE retryWhileAgain #-}
 
+-------------------------------------------------------------------------------
+
+withPrimSafe' :: forall a b. Prim a => a -> (MBA# a -> IO b) -> IO (a, b)
+withPrimSafe' v f = do
+    mpa@(MutablePrimArray mba#) <- newAlignedPinnedPrimArray 1
+    writePrimArray mpa 0 v
+    !b <- f mba#
+    !a <- readPrimArray mpa 0
+    return (a, b)
+{-# INLINE withPrimSafe' #-}
+
```

Comment:
How is this different from [withPrimSafe](https://hackage.haskell.org/package/Z-Data-0.1.1.0/docs/src/Z.Foreign.html#withPrimSafe)?

Reply:
There is no major difference with 'withPrimSafe' but a ByteArray# instead of a pointer